### PR TITLE
Make module swap the main QAT flow again

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #1020
* __->__ #1019

**Summary:** Following https://github.com/pytorch/ao/issues/987, this
commit makes module swap the main QAT flow today. We remove all
tensor subclass fake quantize injection logic since this is not
needed in both the long term and the short term plans for QAT.
In the short term, we will continue to use a full module swap
flow, and only migrate to the long term flow once there is
general distributed support for tensor subclasses and when
tensor subclass composability provides meaningful benefits.

**Test Plan:**
python test/quantization/test_qat.py

### CODE REVIEW

File: torchao/quantization/prototype/qat/api.py [hunk 1]
```diff
@@ -70,207 +47,3 @@ def convert(
         for quantizer in self.quantizers:
             model = quantizer.convert(model)
         return model
-
-
-# =================
-# |   8da4w QAT   |
-# =================
-
-def int8_dynamic_activation_int4_weight_fake_quantize(group_size=32):
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torchao/quantization/prototype/qat/api.py:79]: I thought we decided on using fake quantize tensor subclass for the long term: https://github.com/pytorch/ao/issues/987?

[the remark this answers is not in the document]
Comment (andrewor14) [reply torchao/quantization/prototype/qat/api.py:79]: This path is doing tensor subclass injection. I think we don't want to do that in the long term. E.g. what fp8 training does is use module swaps to insert the subclasses, like: https://github.com/pytorch/ao/blob/dec03131484b5717e9fb8b6c0c25f41ec30e9fef/torchao/float8/float8_linear.py#L500.

[the remark this answers is not in the document]
Comment (jerryzh168) [reply torchao/quantization/prototype/qat/api.py:79]: oh OK I thought `AffineFakeQuantizedTensor` was removed. then sounds good

**Review by jerryzh168 — APPROVED** [review]

**Comment by andrewor14** [comment]
oh no, this was merged into the wrong branch, need to reopen again

**Comment by andrewor14** [comment]
Reopened here: https://github.com/pytorch/ao/pull/1037

**Pull request merged**